overlord/snapstate/refreshhints.go

```diff
@@ -189,3 +189,24 @@ func refreshHintsFromCandidates(st *state.State, updates []*snap.Info, ignoreVal
 	}
 	return hints, nil
 }
+
+// pruneRefreshCandidates removes the given snaps from refresh-candidates map
+// in the state.
+func pruneRefreshCandidates(st *state.State, snaps ...string) error {
+	var candidates map[string]*refreshCandidate
+
+	err := st.Get("refresh-candidates", &candidates)
+	if err == state.ErrNoState {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
nitpick, mainly a personal preference:
```suggestion
	if err != nil {
		if err == state.ErrNoState {
			return nil
		}
		return err
	}
```